File: Sorting/selectionsort.py
# Time Complexity O(n^2)


def selectionsort(arr,n):
    for i in range(0,n-1):
        min=i
        for j in range(i,n):
            if arr[j]<arr[min]:
                min=j
        temp=arr[i]
        arr[i]=arr[min]
        arr[min]=temp

    return arr


n=int(input())

arr=list(map(int,input().split()))[:n]


print(selectionsort(arr,n))

File: ALgorithms/selectionsort.cpp
#include<iostream>
using namespace std;

int min(int arr[],int n)
{
	int minindex=0;
	for(int i=0;i<n;i++)
	{
		if (arr[minindex]>arr[i])
		{
			minindex=i;
		}
	}
	return minindex;
}
void swap(int arr[],int n)
{
	for(int i=0;i<n;i++)
	{
		int k=min(arr,n);
		int temp=arr[i];
		arr[i]=arr[k];
		arr[k]=temp;
	}
	
}

int main()
{
	int arr[]={9,1,5,2,4,10,6,3};
	int n=sizeof(arr)/sizeof(arr[0]);
	swap(arr,n);
		
	
}
